### DISCUSSION TITLE
setup statement_timeout

### DISCUSSION THREAD

**Discussion by ghost** [cybertec-postgresql/pgwatch2#652] [General]
Ah have some  databases monitored by PGwatch2 with a lot of tables.
I have some troubles like this.
`Jun  7 15:07:49 pgwatch2-daemon: 2023/06/07 15:07:49 ERRO DBExecRead: failed to fully process resultset for db_xxx
 sql: SET lock_timeout TO '100ms';SET statement_timeout TO '5s';
Jun  7 15:07:49  pgwatch2-daemon: select
Jun  7 15:07:49  pgwatch2-daemon: (extract(epoch from now()) * 1e9)::int8 as epoch_ns,
Jun  7 15:07:49  pgwatch2-daemon: quote_ident(table_schema)||'.'||quote_ident(table_name) as tag_table,
Jun  7 15:07:49  pgwatch2-daemon: md5((array_agg((c.*)::text order by ordinal_position))::text)
Jun  7 15:07:49  pgwatch2-daemon: from (
Jun  7 15:07:49  pgwatch2-daemon: SELECT current_database()::information_schema.sql_identifier AS table_catalog,
Jun  7 15:07:49  pgwatch2-daemon: nc.nspname::information_schema.sql_identifier AS table_schema,
Jun  7 15:07:49  pgwatch2-daemon: c.relname::information_schema.sql_identifier AS table_name,
Jun  7 15:07:49  pgwatch2-daemon: a.attname::information_schema.sql_identifier AS column_name,
Jun  7 15:07:49  pgwatch2-daemon: a.attnum::information_schema.cardinal_number AS ordinal_position,
Jun  7 15:07:49  pgwatch2-daemon: pg_get_expr(ad.adbin, ad.adrelid)::information_schema.character_data AS column_default,
Jun  7 15:07:49  pgwatch2-daemon: CASE
Jun  7 15:07:49  pgwatch2-daemon: WHEN a.attnotnull OR t.typtype = 'd'::"char" AND t.typnotnull THEN 'NO'::text
Jun  7 15:07:49  pgwatch2-daemon: ELSE 'YES'::text
Jun  7 15:07:49  pgwatch2-daemon: END::information_schema.yes_or_no AS is_nullable,
Jun  7 15:07:49  pgwatch2-daemon: CASE
Jun  7 15:07:49  pgwatch2-daemon: WHEN t.typtype = 'd'::"char" THEN
Jun  7 15:07:49  pgwatch2-daemon: CASE
Jun  7 15:07:49  pgwatch2-daemon: WHEN bt.typelem <> 0::oid AND bt.typlen = '-1'::integer THEN 'ARRAY'::text
Jun  7 15:07:49  pgwatch2-daemon: WHEN nbt.nspname = 'pg_catalog'::name THEN format_type(t.typbasetype, NULL::integer)
Jun  7 15:07:49  pgwatch2-daemon: ELSE 'USER-DEFINED'::text ...
`
If I understand this because the query response if very long for a exécution in 5 secondes.
Can I setup the statement_timeout ?

**Comment by kmoppel** [comment]
@sebastiensophie Btw, how many tables did you have?

**Reply by ghost** [reply]
I have 22478 tables.
I use subpartitions I think those queries take it in count.
For the moment I just increase Statement timeout.
